torch/_export/__init__.py

```diff
@@ -229,90 +219,6 @@ def _eval(self, mode: bool = True):
-
-    assert fake_mode is not None, "Cannot find fake_mode attatched to the graph's placeholders."
-
-    count = 0
-
-    def convert_to_fake(x):
-        nonlocal count
-        val = fake_inps[count]
-        count += 1
-        return val
-
-    fake_args = pytree.tree_map_only(torch.Tensor, convert_to_fake, args)
-    # TODO properly use the cached fake tensor
-    fake_kwargs = pytree.tree_map_only(torch.Tensor, fake_mode.from_tensor, kwargs)
-    fake_params_buffers = pytree.tree_map_only(torch.Tensor,
-                                               functools.partial(fake_mode.from_tensor, static_shapes=True),
-                                               {**dict(gm.named_parameters(remove_duplicate=False)),
-                                                **dict(gm.named_buffers(remove_duplicate=False))})
-    return fake_args, fake_kwargs, fake_params_buffers, fake_mode
-
-
-def _replace_param_buffer_names(param_buffer_table, sig):
-    for spec in sig.input_specs:
-        spec.target = param_buffer_table.get(spec.target, spec.target)
-    for spec in sig.output_specs:
-        spec.target = param_buffer_table.get(spec.target, spec.target)
-
-
-def _normalize_nn_module_stack(gm_torch_level, root_cls):
-    # Append a root module to every nn_module_stack.
-    root = "L['self']"
-    root_key = re.sub(r'[^a-zA-Z0-9]', '_', root)
-    for gm in gm_torch_level.modules():
-        if not isinstance(gm, torch.fx.GraphModule):
-            continue
-        for node in gm.graph.nodes:
-            if node.op in ["placeholder", "output"]:
-                continue
-            add_root = True
-            if nn_module_stack := node.meta.get("nn_module_stack", {}):
-                path, ty = next(iter(nn_module_stack.values()))
-                assert issubclass(ty, torch.nn.Module)
-                # TODO Figure out why sometimes we have root sometimes we don't.
-                if path == root and ty is root_cls:
-                    add_root = False
-            if add_root:
-                def normalize_path(path):
-                    try:
-                        parts = []
-
-                        class Path:
-                            def __getattr__(self, name):
-                                parts.append(name)
-                                return self
-
-                            def __getitem__(self, idx):
-                                parts.append(str(idx))
-                                return self
-
-                        eval(path, {"L": {"self": Path()}})
-                        return ".".join(parts)
-                    except Exception:  # TODO(zhxchen17) Remove this.
-                        return path
-
-                nn_module_stack = {root_key: (root, root_cls), **nn_module_stack}
-                node.meta["nn_module_stack"] = {
-                    key: (normalize_path(path), ty)
-                    for key, (path, ty) in nn_module_stack.items()
-                }
-
 def _export_to_torch_ir(
```

Comment:
why we want to keep this function and reforward to _trace._export_to_torch_ir? Can we just do from torch.export._trace import _export_to_torch_ir and remove this forwarding?

Reply:
Eventually I want to get rid of reliance on torch/_export, and move all calls to torch.export._trace.export_to_torch_ir. Might as well remove it in this diff and do the forwarding in torch._export. Does that make sense?